### PR TITLE
feat: [IDP-1272]: added licence enforcement for IDP

### PR DESCRIPTION
**Pull request by bhavya-sinha**
### Summary
Added swagger json, and generated service using restful react for IDP because of following reason:
In idp-service, backend started following openapi convention for APIs. But for licensing integration, BE leverage platform framework. So all the request and response DTOs lies in platform code.
If OpenAPI is defined, there is a risk of duplication, and maintaining module-related code for platform integration is not advisable. Hence, the decision was made to define the APIs in Swagger.
CC: @sathish-soundarapandian @sunnykesh-harness 

Haven't added coverage in this PR, as we need to urgently merge it for GA. Will add coverage in followup PRs.
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [x] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Review by devinder-harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks good to me

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/IDPUsageGraph.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+          <Layout.Vertical className={pageCss.badgesContainer}>
+            <div>{getSummaryCardRenderers(summaryCardsData)}</div>
+          </Layout.Vertical>
+
+          <Layout.Horizontal spacing="small" flex={{ justifyContent: 'flex-end' }}>
+            <Select items={getLastSixMonths()} value={selectedTimePeriod} onChange={onFilterChange} />
+          </Layout.Horizontal>
+        </Layout.Horizontal>
+        {loading && <PageSpinner />}
+        <StackedColumnChart options={customChartOptions} data={[]}></StackedColumnChart>
```
Comment (jenil-harness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/overview/IDPUsageGraph.tsx:244]: data is empty array, is this expected?

Comment (bhavya-sinha) [reply src/modules/27-platform/auth-settings/pages/subscriptions/overview/IDPUsageGraph.tsx:244]: yes

File: src/modules/27-platform/auth-settings/pages/subscriptions/overview/IDPUsageGraph.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
+/*
+ * Copyright 2024 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { useState } from 'react'
```
Comment (jenil-harness) [inline src/modules/27-platform/auth-settings/pages/subscriptions/overview/IDPUsageGraph.tsx:8]: knit: use moment for dates.

**Review by jenil-harness — APPROVED** [review]
LGTM

**Pull request merged**